pkg/schedule/checker/checker_controller.go

```diff
@@ -67,10 +84,119 @@ func NewController(ctx context.Context, cluster sche.CheckerCluster, conf config
 		priorityInspector:       NewPriorityInspector(cluster, conf),
 		pendingProcessedRegions: pendingProcessedRegions,
 		suspectKeyRanges:        cache.NewStringTTL(ctx, time.Minute, 3*time.Minute),
+		patrolRegionContext:     &PatrolRegionContext{},
+	}
+}
+
+// PatrolRegions is used to scan regions.
+// The checkers will check these regions to decide if they need to do some operations.
+func (c *Controller) PatrolRegions() {
+	c.patrolRegionContext.init(c.cluster)
+	ticker := time.NewTicker(c.cluster.GetCheckerConfig().GetPatrolRegionInterval())
+	defer ticker.Stop()
+	var (
+		key     []byte
+		regions []*core.RegionInfo
+	)
+	for {
+		select {
+		case <-ticker.C:
+			c.updateTickerIfNeeded(ticker)
+		case <-c.ctx.Done():
+			patrolCheckRegionsGauge.Set(0)
+			c.patrolRegionContext.setPatrolRegionsDuration(0)
+			return
+		}
+		if c.cluster.IsSchedulingHalted() {
+			continue
+		}
+
+		// Check priority regions first.
+		c.checkPriorityRegions()
+		// Check pending processed regions first.
+		c.checkPendingProcessedRegions()
+
+		key, regions = c.checkRegions(key)
+		if len(regions) == 0 {
+			continue
+		}
+		// Updates the label level isolation statistics.
+		c.cluster.UpdateRegionsLabelLevelStats(regions)
+		if len(key) == 0 {
+			c.patrolRegionContext.roundUpdateMetrics()
+		}
+		failpoint.Inject("breakPatrol", func() {
+			failpoint.Break()
+		})
+	}
+}
+
+func (c *Controller) updateTickerIfNeeded(ticker *time.Ticker) {
+	// Note: we reset the ticker here to support updating configuration dynamically.
+	newInterval := c.cluster.GetCheckerConfig().GetPatrolRegionInterval()
+	if c.patrolRegionContext.interval != newInterval {
```

Comment:
Why not keep it the same as before, just reset it every time?

Reply:
Maybe move it to patrolRegionContext?